heft-plugins/heft-workbox-plugin/src/WorkboxPlugin.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { ITerminal, Import } from '@rushstack/node-core-library';
+import type {
+  HeftConfiguration,
+  HeftSession,
+  ICompileSubstage,
+  IBuildStageContext,
+  IHeftPlugin,
+  ScopedLogger
+} from '@rushstack/heft';
+
+import { configurationFileLoader, IWorkboxConfigurationJson } from './WorkboxConfigLoader.js';
+
+import { copyFile } from 'fs';
+
+const workboxBuild: typeof import('workbox-build') = Import.lazy('workbox-build', require);
+
+const PLUGIN_NAME: string = 'WorkboxPlugin';
+
+interface IWorkboxConfigurationFileCacheEntry {
+  configurationFile: IWorkboxConfigurationJson | undefined;
+}
+
+/**
+ * @internal
+ */
+export class WorkboxPlugin implements IHeftPlugin {
+  public readonly pluginName: string = PLUGIN_NAME;
+
+  private _workboxConfigurationFileCache: Map<string, IWorkboxConfigurationFileCacheEntry> = new Map<
+    string,
+    IWorkboxConfigurationFileCacheEntry
+  >();
```

Comment:
`heft-config-file` has an internal cache, so you shouldn't need this.

Reply:
I was following the example in the Typescript plugin: https://github.com/microsoft/rushstack/blob/master/apps/heft/src/plugins/TypeScriptPlugin/TypeScriptPlugin.ts#L159

Is this definitely not needed?